compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3474,6 +3474,15 @@ public BType resolveExprType(BType type) {
      * @return boolean whether the type is basic type or not.
      */
     public boolean isBasicType(BType type) {
+        if (type.tag == TypeTags.ARRAY) {
+            BType arrType = ((BArrayType) type).eType;
+            return checkBasicType(arrType);
+        } else {
+            return checkBasicType(type);
```

Comment:
No need to special case the array right? Can use the same switch case to do this.

Reply:
Also, what's the exact purpose of this method? The name of the method conveys a completely different meaning from its implementation. e.g., this seems to consider union types with `()` as basic. But in the spec, unions aren't considered as a basic type.

Even without the union, this seems to be considering an arbitrary set of types from the basic types.

Reply:
Will look into it and do the necessary changes

Reply:
This also seems to miss the int and string subtypes. I also think `isBasicType` (I think `isSimpleBasicType` is more accurate here) should just check for the simple basic types in line with the spec, and we can maybe introduce a separate method that handles arrays and unions, eventually calling `isBasicType`.